src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java

```diff
@@ -107,8 +116,18 @@ public String launch(AppDeploymentRequest request) {
 
 			Map<String, String> jobAnnotations = getJobAnnotations(request);
 
-			if (properties.isCreateJob()) {
-				launchJob(appId, podSpec, podLabelMap, idMap, jobAnnotations);
+			Map<String, String> jobSpecProperties = new HashMap<>();
```

Comment:
given this only applies to `Job`'s, can just do this in `launchJob` passing in the request rather than building the map, etc